### ISSUE TITLE
[BUG] Any command with Spaces fails in OSX Catalina with Python 3.8.5

### ISSUE THREAD

**Issue by sankalpsingha** [calebstewart/pwncat#49]
**Issue Description**
I am trying to get a reverse connection from a box. My current setup of pwncat is in a virtualenv in my OSX Catalina 10.15.6 on Python 3.8.5 

After receiving the connection, two main issues:

1. The prompt is weird. 
2. Anytime I run a command with spaces ( ex. ls -la ) the connection breaks with the error message reported below.

**Target System**
THM box "dogcat" ( I am sure you know about this box @JohnHammond  )

**Error Log**
```
\[\033[01;31m\](remote)\[\033[00m\] \[\033[01;33m\]\u@\h\[\033[00m\]:\[\033[01;36m\]\w\[\033[00m\]$
[01:50:40] local terminal restored                                                                                                                                                                __main__.py:74
Traceback (most recent call last):
  File "/Users/testuser/Documents/CTF/pwncat/venv/bin/pwncat", line 33, in <module>
    sys.exit(load_entry_point('pwncat==0.3.1', 'console_scripts', 'pwncat')())
  File "/Users/testuser/Documents/CTF/pwncat/venv/lib/python3.8/site-packages/pwncat-0.3.1-py3.8.egg/pwncat/__main__.py", line 60, in main
    sys.stdout.flush()
BlockingIOError: [Errno 35] write could not complete without blocking
```

**Screenshots**
<img width="1287" alt="Screen Shot 2020-08-25 at 01 56 35" src="https://user-images.githubusercontent.com/2706955/91084478-41a3ba80-e676-11ea-8a74-3726b13410bc.png">

**ASCII Cinema Recording**
[![asciicast](https://asciinema.org/a/qLvuxkyprdgWopQrHa4DSVt1P.svg)](https://asciinema.org/a/qLvuxkyprdgWopQrHa4DSVt1P)

**Comment by calebstewart** [comment]
The prompt issue is really the same as the `dash` issues mentioned in the readme, but admittedly hadn't been noted there to affect other places. When run as `/bin/sh`, most shells try to act like the unix standard `sh` which doesn't support a lot of interactive features (like fancy colored/dynamic prompts), so the prompt looks weird. 

6a7af17 and 30e084a should fix your issues. I removed the blocking flag from stdio, and added a check/warning for running in `/bin/sh`. You will be defaulted to the basic prompt if the shell was invoked as `/bin/sh`, and there will be a warning printed about it. 

I've tested privilege escalation and enumeration with the non-blocking changes I made and it doesn't appear to break anything. I believe I originally had the non-blocking flag in order to try to send/receive larger chunks of data, but with some of the reorganization that happened relatively recently, I don't believe it's required anywhere now. Let me know if it causes any issues on your end and/or fixes the issue.
